include/boost/geometry/io/wkt/write.hpp

```diff
@@ -515,40 +596,60 @@ class wkt_manipulator
 
     // Boost.Geometry, by default, closes polygons explictly, but not rings
     // NOTE: this might change in the future!
-    inline wkt_manipulator(Geometry const& g,
+    inline wkt_manipulator(Geometry const& g, int significant_digits = -1,
                            bool force_closure = ! is_ring)
         : m_geometry(g)
+        , m_significant_digits(significant_digits)
         , m_force_closure(force_closure)
     {}
 
     template <typename Char, typename Traits>
     inline friend std::basic_ostream<Char, Traits>& operator<<(
-            std::basic_ostream<Char, Traits>& os,
+            std::basic_ostream<Char, Traits>& out,
             wkt_manipulator const& m)
     {
-        dispatch::devarianted_wkt<Geometry>::apply(os, m.m_geometry, m.m_force_closure);
-        os.flush();
-        return os;
+        dispatch::devarianted_wkt<Geometry>::apply(out, m.m_geometry, m.m_significant_digits, m.m_force_closure);
+        out.flush();
+        return out;
     }
 
 private:
     Geometry const& m_geometry;
+    int m_significant_digits;
     bool m_force_closure;
 };
-
 /*!
 \brief Main WKT-streaming function
 \tparam Geometry \tparam_geometry
 \param geometry \param_geometry
+\param significant_digits Specifies the no of significant digits to use in the output wkt
 \ingroup wkt
 \qbk{[include reference/io/wkt.qbk]}
 */
 template <typename Geometry>
-inline wkt_manipulator<Geometry> wkt(Geometry const& geometry)
+inline wkt_manipulator<Geometry> wkt(Geometry const& geometry, int significant_digits = -1)
```

Comment:
This is not needed here, a user can set precision the old way, using `std::setprecision` for streams.
Am I missing some discussion about it?

Reply:
No, I don't think you are. I think it slipped through my review unnoticed

Reply:
In my opinion if we are providing the option to users to set significant digits in `to_wkt` it would seem symmetric to let users set it in `wkt` too. What do you think @awulkiew and @mloskot?

Reply:
@sudo-panda Yes, it implicitly seemed logical and that's why it had not been caught..

I don't think it is a major flaw in design as for some  this may be more convenient to stream's `precision` call:

```cpp
std::cout << wkt(g, n)
```
On the other hand, that interface is not as capable as the stream's interface, it does not allow other manipulators. 

Moreover, we want to avoid confusing users allowing them to do

```cpp
std::cout.precision(n1);
std::cout << wkt(g, n2)
```

That would raise unnecessary questions, wouldn't it?

AFAIU @awulkiew, we don't want to compete with `std::ostream` features.

Reply:
> AFAIU @awulkiew, we don't want to compete with std::ostream features.

@mloskot I think so. It doesn't make sense for me to allow passing the precision here. `bg::wkt` is an IO manipulator similar to `std::setprecision` itself. It has a clear job to do. It is not needed to semantically "merge" both manipulators together especially when we consider that `std::setprecision` is only one of the many that the user can use to tweak the output to his/her needs. See: https://en.cppreference.com/w/cpp/io/manip

IMO the correct usage is:

    std::cout << std::setprecision(n1) << wkt(g);
    // or
    std::cout << std::setprecision(n1) << std::fixed << std::nouppercase << wkt(g);

I rather see the additional parameter of `bg::to_wkt()` as a "convenient workaround", not as an example of "good design" that should be replicated in other places. To the extent that I'd prefer to remove it, only support the default and force the user to use `bg::wkt` if something else was needed. For a long time we didn't even have `to_wkt` because implementing it with `bg::wkt` is trivial if you really need something like that. But I'm ok with this additional parameter for users convenience if you think this would be useful.

Reply:
@awulkiew 
>I rather see the additional parameter of bg::to_wkt() as a "convenient workaround", not as an example of "good design" that should be replicated in other places.

The `to_wkt` is just a simpler function offered for convenience and it has limited capabilities in comparison to the streams. It could be more limited as using default precision or it could be less limited as accepting the extra argument, but nothing else, no other equivalents to the manipulators. I wouldn't call it a bad design, rather a choice, a tradeoff.

> I'm ok with this additional parameter for users convenience if you think this would be useful.

Although the low-level plumbing requires some adjustments to allow the porcelain interface,
I don't think it is a huge interference or it will impose maintenance hassle in future (or does it?). And some users prefer simplified variants of interfaces https://github.com/boostorg/geometry/issues/654#issuecomment-570955873

That's it, there is no deep thoughts behind any of that.

Reply:
> I wouldn't call it a bad design, rather a choice, a tradeoff.

Me neither. I don't think it's good. It doesn't mean that it's bad ;). It's just straightforward, ad-hoc solution, not very elegant. We have several functions like that so I'm ok with it. I'm talking about `to_wkt` here. I still think that `wkt` should not have this parameter.

> I don't think it is a huge interference or it will impose maintenance hassle in future (or does it?)

No, I don't think so.

Reply:
> I still think that wkt should not have this parameter.

Yes, I agree. I think it's a good catch you've made. I admit, it's my bad I missed that is unnecessary indeed.